.github/workflows/cpp.yml

```diff
@@ -45,11 +45,12 @@ concurrency:
   group: ${{ github.repository }}-${{ github.head_ref || github.sha }}-${{ github.workflow }}
   cancel-in-progress: true
 
+permissions:
+  contents: read
+
 env:
   ARROW_ENABLE_TIMING_TESTS: OFF
   DOCKER_VOLUME_PREFIX: ".docker/"
-  ARCHERY_DOCKER_USER: ${{ secrets.DOCKERHUB_USER }}
-  ARCHERY_DOCKER_PASSWORD: ${{ secrets.DOCKERHUB_TOKEN }}
```

Comment:
We may need them for `archery docker  run` to avoid pull rate limit: https://docs.docker.com/docker-hub/download-rate-limit/

> For anonymous users, the rate limit is set to 100 pulls per 6 hours per IP address. For [authenticated](https://docs.docker.com/docker-hub/download-rate-limit/#how-do-i-authenticate-pull-requests) users, it is 200 pulls per 6 hour period.

If it's not a problem for us, we don't need to use them.

Reply:
I have added the secrets to the run stages to, I don't think it is a huge problem but it might cause spurious CI failures and the fix is easy 👍